solr/core/src/java/org/apache/solr/cloud/gumi/CreateCollectionRequest.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.solr.cloud.gumi;
+
+/**
+ * <p>Request for creating a new collection with a given number of shards and replication factor for various replica types.</p>
+ *
+ * Note there is no need at this stage to allow the request to convey each shard hash range for example, this can be handled
+ * by the Solr side implementation without needing the plugin to worry about it.
+ *
+ */
+public interface CreateCollectionRequest extends Request {
+  String getCollectionName();
+
+  int getShardCount();
```

Comment:
Users must be able to request specific shard names, not just the count, eg. when implicit router is used. Other parameters are needed too, such as the router parameters, non-default shard hash ranges etc. See CollectionAdminRequest for details.

Reply:
Shard index can likely go away altogether and we only keep shard names (+ matching routing info).

I'm trying to understand if naming the shards should be the responsibility of the Autoscaling plugin or rather of Solr (would prefer the plugin to always deal with preexisting shard names, makes things simpler).
If it's the responsibility of Solr, when the API collection create call does not specify names, Solr first creates the corresponding names and then calls the plugin.

I'm interested in comments for or against this approach.